sys/include/saul_reg.h

```diff
@@ -55,6 +56,20 @@ typedef struct {
  */
 extern saul_reg_t *saul_reg;
 
+/**
+ * @brief   SAUL complex pointer
+ *
+ * @note    The SAUL registry is comprised of real members and imaginary
+ *          duplicates. If a device is added to the registry, it will add
+ *          imag_count duplicates without using additional system resources.
+ *          This is intended to make support easier for devices with an array
+ *          of similar inputs/outputs (such as GPIO expanders, or analog muxes).
+ */
+typedef struct saul_complex_ptr {
+    struct saul_reg *real;    /**< pointer to real device in SAUL registry */
+    uint8_t imag;             /**< pointer to imaginary device duplicate */
```

Comment:
This is more an index than a pointer right?

Reply:
It is an analogy. It is a "pointer" that points to an "imaginary" device by way of an additional parameter. So far as the SAUL functions are concerned: It is as if there is an imaginary part of memory that holds the (actually nonexistent) data about the device.

Reply:
Ah, IMHO using pointer in the comment is a bit confusing. Thanks for clarifying!